mypyc/lib-rt/list_ops.c

```diff
@@ -79,7 +79,19 @@ bool CPyList_SetItem(PyObject *list, CPyTagged index, PyObject *value) {
         PyList_SET_ITEM(list, n, value);
         return true;
     } else {
-        PyErr_SetString(PyExc_IndexError, "list assignment index out of range");
+        PyErr_SetString(PyExc_OverflowError, "Python int too large to convert to C ssize_t");
```

Comment:
Brainstorming: "int too large for list index" could be a slightly better message? We generally want to avoid C concepts in error messages.